lib/stat_tracker.rb

```diff
@@ -97,6 +97,68 @@ def highest_scoring_home_team
     find_team(:home, "max")
   end
 
+  def winningest_coach(season)
+    game_id_list = []
+    game_list = []
+    @games.each do |game|
+      game_id_list << game.game_id if game.season == season
+    end
+    @game_teams.each do |game|
+      game_list << game if game_id_list.include?(game.game_id)
+    end
+    coach_win_hash = Hash. new(0)
+    games_total_by_coach = Hash.new(0)
+    coach_win_percentage = Hash.new(0.0)
+    winner = nil
+    game_list.each do |game|
+      coach_win_hash[game.head_coach] += 1 if game.result == 'WIN'
+    end
+    game_list.each do |game|
+      games_total_by_coach[game.head_coach] += 1 if game.result != nil
+    end
+    games_total_by_coach.each do |coach, points|
+      coach_win_hash.each do |c, p|
+        coach_win_percentage[c] = p.fdiv(points).round(2) if coach == c
+      end
+    end
+    coach_win_percentage.select do |coach, win_percent|
+      winner = coach if win_percent == coach_win_percentage.values.max #&&
+      # coach_win_hash.values.max == coach_win_hash[coach] #only works for small dataset
```

Comment:
Hey, why is this part commented out?

Reply:
I wasn't using that part for a certain set of data. but I think I need it now!

Reply:
I fixed it!